src/Fabulous/Program.fs

```diff
@@ -94,12 +83,71 @@ type ProgramRunner<'arg, 'model, 'msg>(host: IHost, program: Program<'arg, 'mode
                 host.SetRootView(pageObj)
 
             lastViewDataOpt <- Some newPageElement
-                      
+    let viewInbox = MailboxProcessor.Start (fun inbox ->
+         let rec loop ()  = async{
+             match! inbox.Receive() with
+             | ViewMsg.Render ->
+                 Debug.WriteLine "View invoked"
+                 program.syncAction (fun()->updateView lastModel) ()
+                 return! loop ()
+         }
+         
+         loop ()        
+          )   
+    let inbox=MailboxProcessor.Start(fun inbox ->
```

Comment:
I would not create a mailbox processor for that, because we don't need to wait before computing a new model.

We can keep the existing `processMsg` function to immediately compute a new model.
Only, instead of calling `updateView updatedModel` directly from `processMsg`, we would do `viewInbox.Post(updatedModel)`.

```fs
let rec processMsg msg = 
    try
        let (updatedModel,newCommands) = program.update msg lastModel
        viewInbox.Post(updatedModel)

        for sub in newCommands do
            try 
                sub dispatch
            with ex ->
                program.onError ("Error executing commands:", ex)
    with ex ->
        program.onError ("Unable to process a message:", ex)
```

Then in the `viewInbox`, it would do something like
```fs
let rec loop() =
    let queueSize = inbox.CurrentQueueLength

    if queueSize > 1 then
        // Discard old messages except last
        for i = 1 .. queueSize - 1 do
            let! _ = inbox.Receive 0 // Immediately reads the message and discard it

    let! updatedModel = inbox.Receive()
    lastModel <- updatedModel // Only store it here since we need to have the last model of what's on the screen
    try
        updateView updatedModel
    with
         (...)

    loop()
```
_NB: This example is more like pseudo-code than actual working code_

Reply:
Also I think it's important to make sure `update` is not run on the UI thread as this one will be needed by `updateView`.

Reply:
@TimLariviere , unfortunately, there is no way to drop messages from mailbox queue.

Reply:
inbox.Receive/inbox.TryReceive have a timeout parameter you can set to 0 to immediately pull the message from the queue.
So we can drop message like that.